glommio/src/executor/placement/mod.rs

```diff
@@ -214,23 +248,76 @@ impl CpuSet {
 
     /// Returns a reference to the [`CpuLocation`]s currently included in the
     /// `CpuSet`.
-    pub fn as_vec(&self) -> &Vec<CpuLocation> {
-        &self.0
+    pub fn as_vec(&self) -> Vec<&CpuLocation> {
+        self.0.iter().collect()
     }
 
     /// Checks whether the `CpuSet` is empty.
     pub fn is_empty(&self) -> bool {
-        self.as_vec().is_empty()
+        self.0.is_empty()
+    }
+
+    /// An iterator visiting all [`CpuLocation`]s in this CpuSet in arbitrary order.
+    pub fn iter(&self) -> Iter<'_, CpuLocation> {
+        self.0.iter()
     }
 
     /// Returns the number of CPUs included in the `CpuSet`.
     pub fn len(&self) -> usize {
-        self.as_vec().len()
+        self.0.len()
     }
 
     /// Consumes the `CpuSet` and returns the [`CpuLocation`]s.
-    fn take(self) -> Vec<CpuLocation> {
-        self.0
+    fn take(mut self) -> Vec<CpuLocation> {
+        self.0.drain().collect()
+    }
```

Comment:
A better approach would be to add an `into_iter` function that returns an owning iterator that users can `collect()` themselves into whatever they need, vec or otherwise. That's the more common and flexible approach IMO.

Reply:
Should I remove this function? I refactored the one calling site to not use it. If we want to remove it, we need to do it before `1.0.0`.

Reply:
Either keep this or remove it and add an owning iterator variant. I have a preference for the letter option.